### ISSUE TITLE
bug: frozen pool due to insufficient reserves and LP share supply of 0

### ISSUE THREAD

**Issue by Unique-Divine** [NibiruChain/nibiru#1112]
This issue describes a potential edge case in the `dex` module that we didn't account for. 

# Observed behavior

### Image 1: Current output of nibid q dex get-pool 1  (devnet-2).
```json
{
  "pool": {
    "id": "1",
    "address": "nibi1w00c7pqkr5z7ptewg5z87j2ncvxd88w43ug679",
    "pool_params": {
      "swap_fee": "0.010000000000000000",
      "exit_fee": "0.020000000000000000",
      "A": "0",
      "pool_type": "BALANCER"
    },
    "pool_assets": [
      {
        "token": {
          "denom": "unibi",
          "amount": "1"
        },
        "weight": "53687091200000000"
      },
      {
        "token": {
          "denom": "unusd",
          "amount": "424"
        },
        "weight": "53687091200000000"
      }
    ],
    "total_weight": "107374182400000000",
    "total_shares": {
      "denom": "nibiru/pool/1",
      "amount": "0"
    }
  }
}
```

### Image 2:  Joining the pool using even 1unusd fails with "too few assets to join this pool"

![image](https://user-images.githubusercontent.com/51418232/206310772-e40f8541-0eef-48d1-8cc3-499b41522a98.png)

### Image 3: Joining the pool using just 1unibi also fails with "too few assets to join this pool"

![image](https://user-images.githubusercontent.com/51418232/206310793-33eda750-ad12-45cc-a975-aa8905c3f8ad.png)

### Image 4: Matching the reserve ratio gives an "invalid coins" error since there are 0 pool shares.

![image](https://user-images.githubusercontent.com/51418232/206311760-9acff0e6-b325-49ad-9fa0-5f2fbac44857.png)

### Problem description

In other words, the pool is stuck. I've tried other variations of `nibid tx dex join-pool` and can't seem to get any new shares minted. Since reserves are highly imbalanced, the small amount of `unibi` also creates a problem. I guess that's checkmate for this pool.

# How I created this scenario

1. I created a pool for unibi:unusd with healthy reserves.
2. I swapped away all of the unibi reserves.
3. I also burned all my LP shares for no reason.

# Proposal

Provide a work-around feature that re-mints new shares if there aren't any for a live pool. The case created in this issue could be used for one of the test cases.

**Comment by Unique-Divine** [comment]
The single asset side of it seems to be fixed by this PR 1114
- https://github.com/NibiruChain/nibiru/pull/1114

Reopening the issue to address the case where there are 0 shares in the pool.
